### PR TITLE
feat(visual-editing): add OptimisticDocument commit method

### PR DESCRIPTION
**Pull request by rdunk**
Adds a new method for committing optimistic updates independent of applying patches.

Updates the page builder demo app to include a 3d model example. I will move this to a more appropritate place shortly!

### CODE REVIEW

File: packages/visual-editing/src/ui/optimistic-state/useDocuments.ts [hunk 1]
```diff
@@ -89,6 +90,14 @@ function getDocumentsAndSnapshot<T extends Record<string, any>>(id: string, acto
   return {draftId, publishedId, draftDoc, publishedDoc, snapshot}
 }
 
+function createDocumentCommit<T extends Record<string, any>>(id: string, actor: MutatorActor) {
+  return (): void => {
+    const {draftDoc} = getDocumentsAndSnapshot<T>(id, actor)
+    console.log('send submit!')
```
Comment (georgedoescode) [inline packages/visual-editing/src/ui/optimistic-state/useDocuments.ts:96]: do we want to keep this log?

File: apps/page-builder-demo/src/components/overlays/ExcitingTitleControl.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+'use client'
+
+import {at, set} from '@sanity/mutate'
+import {get} from '@sanity/util/paths'
+import {useDocuments, type OverlayComponent} from '@sanity/visual-editing'
+
+export const ExcitingTitleControl: OverlayComponent = (props) => {
+  const {PointerEvents, node} = props
```
Comment (georgedoescode) [inline apps/page-builder-demo/src/components/overlays/ExcitingTitleControl.tsx:8]: I like this way of handling pointerevents!

File: packages/visual-editing/src/ui/optimistic-state/useDocuments.ts (hunk 1 above)
Comment (rdunk) [reply packages/visual-editing/src/ui/optimistic-state/useDocuments.ts:96]: No! Thank you! 🙇

File: apps/page-builder-demo/src/components/overlays/ProductModelRotationControl.tsx [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        const pitch = (45 / 100) * percentageY
+        deferredPatch({yaw, pitch})
+        deferredCommit()
+      }
+    },
+    [deferredCommit, deferredPatch],
+  )
+
+  const handleClick = useCallback(() => {
+    console.log('COMMIT!?')
```
Comment (georgedoescode) [inline apps/page-builder-demo/src/components/overlays/ProductModelRotationControl.tsx:100]: Do we need this?

**Review by georgedoescode — APPROVED** [review]
Focused on `packages/` changes - but looks good to me!

**Pull request merged**